internal/ledger/store/db.go

```diff
@@ -22,6 +22,15 @@ package store
 type DB interface {
 	Get(key Key) (value []byte, err error)
 	Set(key Key, value []byte) error
+	NewIterator(scope Scope) Iterator
```

Comment:
`MockDB` now fails to compile because we use it in component tests (run ledger tests).
We need to run `go:generate` on the interface and implement a simple, but functionally identical `MockDB` method. Also, we need to add it to compare test in `db_cmp_test.go` to ensure that both implementations are identical.

Reply:
done